app/src/main/java/com/futurice/freesound/feature/search/SearchActivity.java

```diff
@@ -75,55 +84,62 @@ public void unbind() {
         }
     };
 
-    private void setClearSearchVisible(@NonNull final Boolean clearVisible) {
-        Preconditions.checkNotNull(closeButton, "Close Button has not been bound");
+    private void setClearSearchVisible(final boolean clearVisible) {
+        checkNotNull(closeButton, "Close Button has not been bound");
 
         get(closeButton).setVisibility(clearVisible ? View.VISIBLE : View.GONE);
     }
 
     public static void open(@NonNull final Context context) {
         checkNotNull(context, "Launching context cannot be null");
+
         Intent intent = new Intent(context, SearchActivity.class);
         context.startActivity(intent);
     }
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_search);
+        setContentView(layout.activity_search);
         Option.ofObj(savedInstanceState)
               .ifNone(this::addSearchFragment);
 
-        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar_search);
+        Toolbar toolbar = (Toolbar) findViewById(id.toolbar_search);
         setSupportActionBar(toolbar);
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
 
-        searchView = (SearchView) findViewById(R.id.search_view);
+        searchView = get((SearchView) findViewById(id.search_view));
         searchView.setIconified(false);
         closeButton = (ImageView) searchView
                 .findViewById(android.support.v7.appcompat.R.id.search_close_btn);
-        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
-            @Override
-            public boolean onQueryTextSubmit(final String query) {
-                return false;
-            }
 
-            @Override
-            public boolean onQueryTextChange(final String newText) {
-                search(newText);
-                return true;
-            }
-        });
         searchView.setOnCloseListener(() -> {
             searchView.setQuery(SearchViewModel.NO_SEARCH, true);
             return true;
         });
 
     }
 
-    private boolean search(final String query) {
-        searchViewModel.search(query);
-        return true;
+    @NonNull
+    private static Flowable<String> getTextChange(@NonNull final SearchView view) {
+        return Flowable.create(new FlowableOnSubscribe<String>() {
```

Comment:
This is much nicer and reactive than my side-effect version. Do you think there's value in making this a utility, our own RxAndroid2 starter-kit? We could well have a few iterations on these ideas while RxJava2 is new, it could be worthwhile to avoid ending up with old patterns in different parts of the code.